app-typescript/components/with-pagination.spec.tsx

```diff
@@ -34,35 +34,68 @@ export class Paginated extends React.PureComponent {
     }
 }
 
-describe.only('with-pagination', () => {
-    it('returns button1 when rendered', (done) => {
+describe('getPagination', () => {
+    it('returns empty array when we have 0 pages', () => {
+        assert.strictEqual(getPagination(1, 0).length, 0);
+    });
+
+    it('returns empty array when we try to access page <= 0', () => {
+        assert.strictEqual(getPagination(-10, 100).length, 0);
+    });
+
+    it('returns dots when on page 1 having 5 total pages', () => {
+        assert.strictEqual(getPagination(1, 5).filter((x) => x === 'dots').length, 1);
+    });
+
+    it('returns dots when on the last page having 5 total pages', () => {
+        assert.strictEqual(getPagination(5, 5).filter((x) => x === 'dots').length, 1);
+    });
+
+    it('returns dots twice when on the middle page', () => {
+        assert.strictEqual(getPagination(5, 10).filter((x) => x === 'dots').length, 2);
+    });
+
+    it('returns page 1 and last page when on the middle page', () => {
+        const res = getPagination(5, 10);
+        const includesFirstAndLast = res[0] === 1 && res[res.length - 1] === 10;
+        assert.strictEqual(includesFirstAndLast, true);
+    });
```

Comment:
can you check the entire output?
`expect getPagination(5, 10) to be [1, 2, 3, '...']`  (pseudo code here)